x-pack/plugins/transform/public/app/sections/transform_management/components/action_edit/use_edit_action.tsx

```diff
@@ -5,25 +5,70 @@
  * 2.0.
  */
 
-import React, { useContext, useMemo, useState } from 'react';
+import React, { useCallback, useContext, useMemo, useState } from 'react';
 
-import { TransformConfigUnion } from '../../../../../../common/types/transform';
+import { i18n } from '@kbn/i18n';
 
 import { TransformListAction, TransformListRow } from '../../../../common';
 import { AuthorizationContext } from '../../../../lib/authorization';
 
 import { editActionNameText, EditActionName } from './edit_action_name';
+import { useSearchItems } from '../../../../hooks/use_search_items';
+import { useAppDependencies, useToastNotifications } from '../../../../app_dependencies';
+import { TransformConfigUnion } from '../../../../../../common/types/transform';
 
 export const useEditAction = (forceDisable: boolean, transformNodes: number) => {
   const { canCreateTransform } = useContext(AuthorizationContext).capabilities;
 
   const [config, setConfig] = useState<TransformConfigUnion>();
   const [isFlyoutVisible, setIsFlyoutVisible] = useState(false);
+  const [indexPatternId, setIndexPatternId] = useState<string | undefined>();
+
   const closeFlyout = () => setIsFlyoutVisible(false);
-  const showFlyout = (newConfig: TransformConfigUnion) => {
-    setConfig(newConfig);
-    setIsFlyoutVisible(true);
-  };
+
+  const { getIndexPatternIdByTitle, loadIndexPatterns } = useSearchItems(undefined);
+  const toastNotifications = useToastNotifications();
+  const appDeps = useAppDependencies();
+  const savedObjectsClient = appDeps.savedObjects.client;
+  const indexPatterns = appDeps.data.indexPatterns;
+
+  const clickHandler = useCallback(
+    async (item: TransformListRow) => {
+      try {
+        await loadIndexPatterns(savedObjectsClient, indexPatterns);
```

Comment:
do we actually need to load _all_ index patterns on every click (opening of the flyout)?

Reply:
I think loading all index patterns is the current standard approach unfortunately. But for this use case, we actually don't need to load it at all since it should already been loaded before the user hits the action. Removed it here https://github.com/elastic/kibana/pull/108804/commits/64375a9b2abc6cbef24de41ba7c48abb57a08a54